src/libraries/Common/src/System/Net/Mail/DotAtomReader.cs

```diff
@@ -29,17 +29,19 @@ internal static class DotAtomReader
         // - -1 if the dot-atom section terminated at the start of the data string.
         //   e.g. "user.name@domain.com", starting at index 8 (e) returns -1.
         //
-        // A FormatException will be thrown if:
+        // Throws FormatException or returns false if:
         // - The initial character at data[index] is invalid in a dot-atom.
         //   e.g. "a@b.com", starting at index 1 (@).
         // - The leading character is a dot.
         //   e.g. "a@.b.com", starting at index 7 (m), throws because the leading char (index 2) is a dot.
         //
-        internal static int ReadReverse(string data, int index)
+        internal static bool TryReadReverse(string data, int index, out int outIndex, bool throwExceptionIfFail)
         {
             Debug.Assert(0 <= index && index < data.Length, "index was outside the bounds of the string: " + index);
 
+            outIndex = default;
```

Comment:
Same here.

Reply:
Waiting to know by @stephentoub if there is some guidance.